### DISCUSSION TITLE
Layout

### DISCUSSION THREAD

**Discussion by stevoh6** [Blackymas/NSPanel_HA_Blueprint#286] [Ideas]
Hi, awesome work :) Im trying do some customizations, mostly Im trying figured out the concept.

Im using v3.1.0 and I would love to develop the energy screen, I already have some ideas (actual, dayily, monthly.....), I already started.

But I would like to open discussion about layout. Im missing one key thing = diferrent home pages - some people would like to have buttons as home screen, some solar production, etc. 

My idea is following. 4diferrent areas (top, bottom, left, right). Top= energy home,  Bottom =free text ( i will explain later) or climate home, Left= buttons home , Right = entities home. 

If you swipe there from main screeen, it will stay there forever until you press "home"/exit  button. From each area home page you can swipe to diferrent direction as it is now and it will return back after some time to area home.  

If you press "entities" button on home screen it will go to entities home page but this time it will return back to main home after some time. Again from entities home you can go to different directions ( entitypage1, entitypage2,3,4) and after some time it will return to main home. 

What do you think?

**Comment by Blackymas** [comment]
We have also thought about adding the start page as an option to the blueprint.

But have not really thought about it yet

**Reply by deejaybeam** [reply]
In v3.1.x you have the possibility to set the page-timeout via ESPHome Device.
If this is set to 0 then it does not jump back to the home-page anymore.
This way you could set each page you want with swiping or via service call and it will stay there.
Not fully tested if there are some side-effects, but it should work this way.

**Comment by edwardtfn** [comment]
- #1718

**Comment by edwardtfn** [comment]
Hi, we are having a discussion about new design for the panel and would be nice to get your comments there: https://discord.com/channels/1464682227068698724/1479900732273721514
